External/Plugins/CodeRefactor/PluginMain.cs

```diff
@@ -322,15 +311,15 @@ private void UpdateMenuItems()
             {
                 this.refactorMainMenu.DelegateMenuItem.Enabled = false;
                 this.refactorContextMenu.DelegateMenuItem.Enabled = false;
-                bool langIsValid = GetLanguageIsValid();
+                bool langIsValid = RefactoringHelper.GetLanguageIsValid();
                 ResolvedContext resolved = ASComplete.CurrentResolvedContext;
                 bool isValid = langIsValid && resolved != null && resolved.Position >= 0;
                 ASResult result = isValid ? resolved.Result : null;
                 if (result != null && !result.IsNull())
                 {
-                    bool isRenameable = (result.Member != null && result.Member.InFile != null && File.Exists(result.Member.InFile.FileName))
-                        || (result.Type != null && result.Type.InFile != null && File.Exists(result.Type.InFile.FileName))
-                        || (result.InFile != null && File.Exists(result.InFile.FileName))
+                    bool isRenameable = (result.Member != null && result.Member.InFile != null && File.Exists(result.Member.InFile.FileName) && !RefactoringHelper.IsUnderSDKPath(result.Member.InFile.FileName))
+                        || (result.Type != null && result.Type.InFile != null && File.Exists(result.Type.InFile.FileName) && !RefactoringHelper.IsUnderSDKPath(result.Type.InFile.FileName))
+                        || (result.InFile != null && File.Exists(result.InFile.FileName) && !RefactoringHelper.IsUnderSDKPath(result.InFile.FileName))
```

Comment:
`IsUnderSDKPath` could take care of verifying if the value isn't null and exists - that would be more readable. Also you could have a little function that accepts a `FileModel`